app/ota_client.py

```diff
@@ -50,73 +49,91 @@ def verify_file(filename: Path, filehash: str, filesize) -> bool:
     return file_sha256(filename) == filehash
 
 
-def _download(url_base: str, path: str, dst: Path, digest: str, *, cookies):
-    quoted_path = urllib.parse.quote(path)
-    url = urllib.parse.urljoin(url_base, quoted_path)
-
-    error_count = 0
-    last_error = ""
-
-    def retry_if_possible(e):
-        nonlocal error_count
-        nonlocal last_error
-        error_count += 1
-        do_retry = True
-        if isinstance(e, requests.exceptions.Timeout):
-            last_error = f"requests timeout: {e},{url=} ({error_count})"
-        elif isinstance(e, requests.exceptions.ConnectionError):
-            last_error = f"requests connection error: {e},{url=} ({error_count})"
-        elif isinstance(e, requests.exceptions.ChunkedEncodingError):
-            last_error = f"requests ChunkedEncodingError: {url=} ({error_count})"
-        elif isinstance(e, RequestException):
-            status_code = getattr(e.response, "status_code", None)
-            last_error = f"requests error: {status_code=},{url=} ({error_count})"
-            if isinstance(status_code, int) and status_code // 100 == 4:
-                do_retry = False
-        else:
-            last_error = f"requests error unknown: {e},{url=}, ({error_count})"
-        logger.warning(last_error)
-        return do_retry
-
-    @retry(
-        stop_max_attempt_number=5,
-        retry_on_exception=retry_if_possible,
-        wait_exponential_multiplier=1_000,
-        wait_exponential_max=10_000,
-    )
-    def _requests_get():
-        CHUNK_SIZE = 4096
-        headers = {}
-        headers["Accept-encording"] = "gzip"
```

Comment:
Ah, typo? `Accept-encoding`

Reply:
Actually the `requests` module will automatically send this header to the server, so don't need to send it by ourself.